### DISCUSSION TITLE
Disable authorisation in fastfile?

### DISCUSSION THREAD

**Discussion by GioPalusa** [fastlane/fastlane#22188] [Q&A]
I'm using Fastlane with Azure Pipelines, and during the summer it stopped working, and I've tried (I think) almost everything to get it to start working again.

Fastlane produces this git command
`git clone https://[TOKEN]@dev.azure.com/[COMPANY]/[PROJECT]/_git/Certificates /var/folders/dx/1mr3dwvd695dgn5gtrmhlwb80000gn/T/d20240815-3903-hw3k7v -c http.extraheader='Authorization: Bearer [TOKEN]'`

this doesn't work any more with the authentication, so I thought I should change to a PAT, but I can only get the git command to work if remove the `-c http.extraheader='Authorization: Bearer [TOKEN]'` part. But I have no idea how, even if I change to `        git_bearer_authorization: nil`. I have also tried the username and password approach with Base64 encoding I used previously, with no luck. 

This pipeline has worked flawlessly for multiple years, so it's just the authentication that I need to fix.

is there any way to change so Fastlane doesn't add the authentication at the end of the git command?

**Comment by jennie258fitz** [comment]
Here are a few steps you can take to resolve the authentication problem:

Switch to Personal Access Token (PAT):
Since the Bearer token method is causing issues, switching to a PAT is a good approach. Ensure you generate a PAT with the necessary permissions for your repository.
Modify Fastlane Configuration: [my AARP Medicare](https://www.myaarpmedicarehealth.com)
You can try to modify the Fastlane configuration to avoid adding the -c http.extraheader='Authorization: Bearer [TOKEN]' part. One way to do this is by setting the git_bearer_authorization to nil in your Fastlane configuration.
Use Environment Variables:
Store your PAT in an environment variable and use it in your Fastlane script. This can help manage the token securely and avoid hardcoding it in your scripts.
Custom Git Command:
If Fastlane doesn’t provide a direct way to modify the git command, you can create a custom script to handle the git clone operation and call this script from your Fastlane configuration.
Then, call this script from your Fastlane configuration:

Ruby

lane :custom_clone do
  sh "./custom_git_clone.sh"
end
Check Fastlane and GitHub Issues:
There might be ongoing discussions or solutions in the Fastlane GitHub issues or forums. For example, you can refer to this discussion for more insights
